### ISSUE TITLE
Possible to use Bolero without Elmish?

### ISSUE THREAD

**Issue by IvanRainbolt** [fsbolero/Bolero#150]
Is it possible to run without Elmish? Would anyone have a simple example of how to set this up and get some html output?

**Comment by OnurGumus** [comment]
Bolero is a regular blazor component . if you don't want to use elmish just create a component with F#.  Bolero is elmish itself.

**Comment by Tarmil** [comment]
Yeah, if you start from the project template, you can replace the entire `Main.fs` with a single component inheriting from `Component` instead of `ProgramComponent`:

```fsharp
type MyApp() =
    inherit Component()

    override this.Render() =
        // Your HTML here
```

**Comment by AlexBoehm** [comment]
> Bolero is a regular blazor component . if you don't want to use elmish just create a component with F#. Bolero is elmish itself.

Does this also mean that you could write Blazor Components in F# which use Elmish internally and can be used like any "regular" component from C#?

**Comment by profK** [comment]
So im trying to do this, but my component executes on the server.  How do I config Bolero to run my client code in WASM on the client?

**Comment by Tarmil** [comment]
@profK Do you mean that you have some side-effect that you run inside `Render()` before returning HTML, that needs to run on the client? Then you need to run this side-effect in Blazor's AfterRender, otherwise it will also run during prerendering on the server.

```fsharp
type MyApp() =
    inherit Component()

    override this.Render() =
        // Your HTML here

    override this.OnAfterRenderAsync(firstRender: bool) =
        if firstRender then
            // Your side-effect here
        base.OnAfterRenderAsync(firstRender)
```

**Comment by profK** [comment]
Thanks but no.  My issue was getting the Client code to run in wasm-hosted mode.  This is not really documented.

In the end, I discovered the flags to bolero-app -m true -s false -r false.

This resulted in a pure client side project to work with.

**Comment by Tarmil** [comment]
The default settings for bolero-app also create a wasm project, albeit with a server side for hosting eg remoting. But yes, if you want it hosted in a plain HTML file, then `-m true -s false` is the way to go.
